flake.nix

```diff
@@ -0,0 +1,31 @@
+{
+  description = "Flake for Plutus";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=7d71001b796340b219d1bfa8552c81995017544a";
+    haskell-nix.url = "github:input-output-hk/haskell.nix?rev=87084d65a476cc826a0e8c5d281d494254f5bc7a";
+    flake-utils.url = "github:numtide/flake-utils?rev=b543720b25df6ffdfcf9227afafc5b8c1fabfae8";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, flake-utils, ... }:
+    (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        topLevel = import ./. {
+          inherit system;
+          isInFlake = true;
+          haskellNix = haskell-nix;
```

Comment:
I think this haskell.nix is not overriding its hackage.nix and stackage.nix to point to specific revisions as we are. Could be why you get different outputs. C.f. https://github.com/input-output-hk/plutus/pull/3221/files#diff-245563af9e34754739b12b294f3324c1d42c116ace7ce8735e21dfbc82d8e064R16